### PR TITLE
feat: add autoapi docs usage

### LINKED ISSUE #361: Ugly and Unhelpful Class Parameters in API docs
image

The class parameters here serve limited purpose ( at least to me ) as they are untyped, are not a link to the relevant place in the docs, and are almost always truncated anyway. Additionally the square bracket syntax for optional arguments is not, I think, an official python thing?

Can we suppress these and just leave the class name here with a link to the docs page?

### PR DESCRIPTION
**Pull request by RobPasMue**
Closes #361

@Revathyvenugopal162 @jorgepiloto - please review thoroughly and feel free to adapt as you prefer.

### CODE REVIEW

File: doc/source/user_guide/autoapi.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    ansys-sphinx-theme[autoapi]
+
+If you are using a ``pyproject.toml`` file, you would define your dependency as follows:
+
+.. code:: toml
+    
+    # For a typical PyAnsys pyproject.toml file
+    [project.optional-dependencies]
+    doc = [
+        "ansys-sphinx-theme[autoapi]==XX.YY.ZZ",
```
Comment (jorgepiloto) [inline doc/source/user_guide/autoapi.rst:30]:
```suggestion
        "ansys-sphinx-theme[autoapi]==X.Y.Z",
```

File: doc/source/user_guide/autoapi.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
+.. _ref_user_guide_autoapi:
+
+Using ``sphinx-autoapi`` to generate API documentation
+------------------------------------------------------
+
+``sphinx-autoapi`` is a tool that automatically generates API documentation from your source code.
```
Comment (jorgepiloto) [inline doc/source/user_guide/autoapi.rst:6]: Suggesting to link the official sphinx-autoapi project just in this paragraph:

```suggestion
`sphinx-autoapi <https://github.com/readthedocs/sphinx-autoapi/>`_ is a tool that automatically generates API documentation from your source code.
```

File: doc/source/user_guide/autoapi.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+It is a great way to keep your documentation up-to-date with your codebase.
+
+The ``ansys-sphinx-theme`` includes a custom template for ``sphinx-autoapi`` that is designed
```
Comment (jorgepiloto) [inline doc/source/user_guide/autoapi.rst:9]:
```suggestion
The ``ansys-sphinx-theme`` include a set of custom templates for ``sphinx-autoapi`` that is designed
```

File: doc/source/user_guide/autoapi.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+Dependencies required
+~~~~~~~~~~~~~~~~~~~~~
+
+To use ``sphinx-autoapi`` with the ``ansys-sphinx-theme``, you need to install the ``autoapi`` target
+of the theme. This can be done by defining your ``ansys-sphinx-theme`` dependency as
+``ansys-sphinx-theme[autoapi]`` in your project's requirements location.
+
+For example, if you are using a ``requirements.txt`` file, you would define your dependency as follows::
+
+    ansys-sphinx-theme[autoapi]
```
Comment (jorgepiloto) [inline doc/source/user_guide/autoapi.rst:21]: Just to be consistent with later lines:

```suggestion
.. code-block: text

    ansys-sphinx-theme[autoapi]==X.Y.Z
```

File: doc/source/user_guide/autoapi.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    autoapi_python_use_implicit_namespaces = True
+    autoapi_keep_files = True
+    autoapi_own_page_level = "class"
+
+
+The above configuration will generate API documentation for your package in the ``src`` directory
+and place the generated files in the ``api`` directory. The generated files will be based on the
+``ansys-sphinx-theme`` templates. If you want to customize some of the previous options, you can
+do so by modifying the configuration above. The main line of code to use the shipped
+templates is the following.
```
Comment (jorgepiloto) [inline doc/source/user_guide/autoapi.rst:68]:
```suggestion
The above configuration generates the API documentation for your package in the ``src`` directory
and places the generated files in the ``api`` directory. These files are based on the
``ansys-sphinx-theme`` templates. If you want to customize some of the previous options, you can
do so by modifying the configuration above. The line of code declaring the desired set of templates is the following one:
```

**Review by jorgepiloto — COMMENTED** [review]
Just left some minor suggestions. Thanks for addressing this.

File: doc/source/user_guide/autoapi.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+to match the look and feel of the rest of the PyAnsys documentation.
+
+Dependencies required
+~~~~~~~~~~~~~~~~~~~~~
+
+To use ``sphinx-autoapi`` with the ``ansys-sphinx-theme``, you need to install the ``autoapi`` target
+of the theme. This can be done by defining your ``ansys-sphinx-theme`` dependency as
+``ansys-sphinx-theme[autoapi]`` in your project's requirements location.
+
+For example, if you are using a ``requirements.txt`` file, you would define your dependency as follows::
```
Comment (RobPasMue) [inline doc/source/user_guide/autoapi.rst:19]:
```suggestion
For example, if you are using a ``requirements.txt`` file, you would define your dependency as follows:
```

**Comment by RobPasMue** [comment]
I need an approving review @ansys/pyansys-core whenever you feel like it - thank you @jorgepiloto for reading it through!

**Review by Revathyvenugopal162 — APPROVED** [review]
LGTM, Thank you, @RobPasMue

**Pull request merged**